front/app/containers/Admin/reporting/components/ReportBuilder/Widgets/SingleIdeaWidget/Settings/index.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useCallback } from 'react';
+
+// hooks
+import { useNode } from '@craftjs/core';
+import usePhases from 'api/phases/usePhases';
+
+// components
+import { Box, Toggle, colors } from '@citizenlab/cl2-component-library';
+import ProjectFilter from '../../_shared/ProjectFilter';
+import PhaseFilter from '../../_shared/PhaseFilter';
+import CollapseLongTextToggle from './CollapseLongTextToggle';
+import InputMultilocWithLocaleSwitcher from 'components/UI/InputMultilocWithLocaleSwitcher';
+
+// i18n
+import messages from '../messages';
+import widgetMessages from '../../messages';
+import { useIntl } from 'utils/cl-intl';
+
+// typings
+import { Props } from '../typings';
+import { IOption, Multiloc } from 'typings';
+import IdeaSelect from 'components/UI/IdeaSelect';
+import { IIdeaData } from 'api/ideas/types';
+
+const Settings = () => {
+  const { formatMessage } = useIntl();
+
+  const {
+    actions: { setProp },
+    title,
+    showAuthor,
+    showContent,
+    showReactions,
+    showVotes,
+    collapseLongText,
+    projectId,
+    phaseId,
+    ideaId,
+  } = useNode<Props>((node) => ({
+    title: node.data.props.title,
+    showAuthor: node.data.props.showAuthor,
+    showContent: node.data.props.showContent,
+    showReactions: node.data.props.showReactions,
+    showVotes: node.data.props.showVotes,
+    collapseLongText: node.data.props.collapseLongText,
+    projectId: node.data.props.projectId,
+    phaseId: node.data.props.phaseId,
+    ideaId: node.data.props.ideaId,
+  }));
+
+  const { data: phases } = usePhases(projectId);
+
+  const setTitle = useCallback(
+    (value: Multiloc) => {
+      setProp((props: Props) => {
+        props.title = value;
+      });
+    },
+    [setProp]
+  );
+
+  const handleChangeShowAuthor = useCallback(
+    (showAuthor: boolean) => {
+      setProp((props: Props) => {
+        props.showAuthor = showAuthor;
+      });
+    },
+    [setProp]
+  );
+
+  const handleChangeShowDescription = useCallback(
+    (showContent: boolean) => {
+      setProp((props: Props) => {
+        props.showContent = showContent;
+      });
+    },
+    [setProp]
+  );
+
+  const handleChangeShowReactions = useCallback(
+    (showReactions: boolean) => {
+      setProp((props: Props) => {
+        props.showReactions = showReactions;
+      });
+    },
+    [setProp]
+  );
+
+  const handleChangeShowVotes = useCallback(
+    (showVotes: boolean) => {
+      setProp((props: Props) => {
+        props.showVotes = showVotes;
+      });
+    },
+    [setProp]
+  );
+
+  const handleChangeCollapseLongText = useCallback(
+    (collapseLongText: boolean) => {
+      setProp((props: Props) => {
+        props.collapseLongText = collapseLongText;
+      });
+    },
+    [setProp]
+  );
+
+  const handleProjectFilter = useCallback(
+    ({ value }: IOption) => {
+      setProp((props: Props) => {
+        props.projectId = value;
+        props.phaseId = undefined;
+      });
```

Comment:
```suggestion
        props.ideaId = undefined;
      });
```

> If you select an idea and then change the project that does not have this idea, the IdeaSelect still shows the old value. But this idea is not shown in the widget, so I guess it's clear to the user that the idea is not actually selected.

Does this suggestion solve this issue?

Reply:
Indeed it does. Thank you! I also reset `ideaId` when the phase changes https://github.com/CitizenLabDotCo/citizenlab/pull/7004/commits/d0e19000c2885601f97e305ca256fbd12995b08e#diff-3d5d5ea76817cc234143dc2f4e3b3fc2675dcc4620187e27d59faecfc01ac18cR122